### ISSUE TITLE
jruby-complete can not install executable-hooks

### ISSUE THREAD

**Issue by mkristian** [jruby/jruby#3557]
```
$ GEM_HOME=asd java -jar ../../jruby/maven/jruby-complete/target/jruby-complete-9.0.5.0-SNAPSHOT.jar -S gem install executable-hooks
Fetching: executable-hooks-1.3.2.gem (100%)
Building native extensions.  This could take a while...
ERROR:  Error installing executable-hooks:
	ERROR: Failed to build gem native extension.

    java -cp :/Users/cmeier/projects/active/jruby/maven/jruby-complete/target/jruby-complete-9.0.5.0-SNAPSHOT.jar org.jruby.Main -r ./siteconf20151221-67692-6iics6.rb extconf.rb
#jar-dependencies 0.2.3 not found in GEM_PATH
#racc 1.4.13 not found in GEM_PATH
rake 10.1.0
Unhandled Java exception: java.lang.NullPointerException
java.lang.NullPointerException: null
                write at org/jruby/util/io/PosixShim.java:102
      flushBufferSync at org/jruby/util/io/OpenFile.java:572
     flushBufferSync2 at org/jruby/util/io/OpenFile.java:563
    flushBufferAsync2 at org/jruby/util/io/OpenFile.java:556
          flushBuffer at org/jruby/util/io/OpenFile.java:549
            io_fflush at org/jruby/util/io/OpenFile.java:425
             finalize at org/jruby/util/io/OpenFile.java:859
              cleanup at org/jruby/util/io/OpenFile.java:802
            rbIoClose at org/jruby/RubyIO.java:2007
            rbIoClose at org/jruby/RubyFile.java:282
                close at org/jruby/RubyIO.java:1955
                 call at org/jruby/internal/runtime/methods/JavaMethod.java:955
       finvokeChecked at org/jruby/RubyClass.java:643
       finvokeChecked at org/jruby/RubyClass.java:625
        invokeChecked at org/jruby/runtime/Helpers.java:446
      checkCallMethod at org/jruby/RubyBasicObject.java:359
              ioClose at org/jruby/RubyIO.java:1965
     ensureYieldClose at org/jruby/RubyIO.java:1134
                 open at org/jruby/RubyIO.java:1125
         cacheAndCall at org/jruby/runtime/callsite/CachingCallSite.java:273
            callBlock at org/jruby/runtime/callsite/CachingCallSite.java:79
                 call at org/jruby/runtime/callsite/CachingCallSite.java:83
            interpret at org/jruby/ir/instructions/CallBase.java:419
          processCall at org/jruby/ir/interpreter/InterpreterEngine.java:322
            interpret at org/jruby/ir/interpreter/StartupInterpreterEngine.java:77
      INTERPRET_BLOCK at org/jruby/ir/interpreter/Interpreter.java:122
      commonYieldPath at org/jruby/runtime/MixedModeIRBlockBody.java:136
              doYield at org/jruby/runtime/IRBlockBody.java:152
                yield at org/jruby/runtime/BlockBody.java:78
                yield at org/jruby/runtime/Block.java:147
              collect at org/jruby/RubyArray.java:2286
                map19 at org/jruby/RubyArray.java:2300
                 call at org/jruby/internal/runtime/methods/JavaMethod.java:494
         cacheAndCall at org/jruby/runtime/callsite/CachingCallSite.java:273
            callBlock at org/jruby/runtime/callsite/CachingCallSite.java:79
                 call at org/jruby/runtime/callsite/CachingCallSite.java:83
            interpret at org/jruby/ir/instructions/CallBase.java:419
          processCall at org/jruby/ir/interpreter/InterpreterEngine.java:322
            interpret at org/jruby/ir/interpreter/StartupInterpreterEngine.java:77
            interpret at org/jruby/ir/interpreter/InterpreterEngine.java:83
     INTERPRET_METHOD at org/jruby/internal/runtime/methods/MixedModeIRMethod.java:197
                 call at org/jruby/internal/runtime/methods/MixedModeIRMethod.java:183
                 call at org/jruby/internal/runtime/methods/DynamicMethod.java:197
                 call at org/jruby/runtime/callsite/CachingCallSite.java:161
          processCall at org/jruby/ir/interpreter/InterpreterEngine.java:290
            interpret at org/jruby/ir/interpreter/StartupInterpreterEngine.java:77
      INTERPRET_BLOCK at org/jruby/ir/interpreter/Interpreter.java:122
      commonYieldPath at org/jruby/runtime/MixedModeIRBlockBody.java:136
              doYield at org/jruby/runtime/IRBlockBody.java:152
                yield at org/jruby/runtime/BlockBody.java:78
                yield at org/jruby/runtime/Block.java:147
                 each at org/jruby/RubyArray.java:1560
                 call at org/jruby/internal/runtime/methods/JavaMethod.java:494
         cacheAndCall at org/jruby/runtime/callsite/CachingCallSite.java:273
            callBlock at org/jruby/runtime/callsite/CachingCallSite.java:79
                 call at org/jruby/runtime/callsite/CachingCallSite.java:83
            interpret at org/jruby/ir/instructions/CallBase.java:419
          processCall at org/jruby/ir/interpreter/InterpreterEngine.java:322
            interpret at org/jruby/ir/interpreter/StartupInterpreterEngine.java:77
            interpret at org/jruby/ir/interpreter/InterpreterEngine.java:77
     INTERPRET_METHOD at org/jruby/internal/runtime/methods/MixedModeIRMethod.java:162
                 call at org/jruby/internal/runtime/methods/MixedModeIRMethod.java:148
                 call at org/jruby/internal/runtime/methods/DynamicMethod.java:189
         cacheAndCall at org/jruby/runtime/callsite/CachingCallSite.java:293
                 call at org/jruby/runtime/callsite/CachingCallSite.java:131
                <top> at extconf.rb:19
  invokeWithArguments at java/lang/invoke/MethodHandle.java:627
                 load at org/jruby/ir/Compiler.java:111
            runScript at org/jruby/Ruby.java:821
            runScript at org/jruby/Ruby.java:813
          runNormally at org/jruby/Ruby.java:751
          runFromMain at org/jruby/Ruby.java:573
        doRunFromMain at org/jruby/Main.java:409
          internalRun at org/jruby/Main.java:304
                  run at org/jruby/Main.java:233
                 main at org/jruby/Main.java:200


extconf failed, exit code 1

Gem files will remain installed in /Users/cmeier/projects/active/lookout/device-config/asd/gems/executable-hooks-1.3.2 for inspection.
Results logged to /Users/cmeier/projects/active/lookout/device-config/asd/extensions/universal-java-1.8/2.2.0/executable-hooks-1.3.2/gem_make.out
```

**Comment by mkristian** [comment]
jruby-1.7.x works

**Comment by headius** [comment]
@mkristian We want to get 9.0.5.0 out, so I'm going to punt to 9.1, the next likely release. If you still want to fix this for 9.0.5.0 let us know.

**Comment by mkristian** [comment]
closing since this is an upstream problem
